人工智能/人工智能数学基础/人智数学基础实验/2026春实验三/实验三代码/newton.py

```
@@ -0,0 +1,213 @@
+    grads = torch.autograd.grad(loss, params, create_graph=True)[0]
+    # Hessian 矩阵 (2x2)
+    hess = torch.zeros(2, 2, dtype=params.dtype)
+    for i in range(2):
+        grad_i = grads[i]
+        hess_i = torch.autograd.grad(grad_i, params, retain_graph=True)[0]
+        hess[i] = hess_i
+    # 正则化
+    hess_reg = hess + reg * torch.eye(2)
+    try:
+        step = torch.linalg.solve(hess_reg, grads)  # 解 H * step = g
+    except:
```

Ruff (0.15.17)

[error] Do not use bare `except`

(E722)